octopusdeploy/schema_deployment_action.go

```diff
@@ -244,6 +244,12 @@ func getActionSchema() (*schema.Schema, *schema.Resource) {
 				Type:             schema.TypeMap,
 				ValidateDiagFunc: warnIfIncludesRunOnServer(),
 			},
+			"sort_order": {
+				Description: "Order used by terraform to ensure correct ordering of actions. This property must be either omitted from all actions, or provided on all actions",
```

Comment:
Given the order is random, would it make more sense to enforce this if `actions > 1` ?

Reply:
yes, breaking change, but IMHO, it is way safer, and hence worth breaking it

Reply:
🤔 part of why i left it is TF has really bad capacity for custom validation. If you run `terraform validate` it seems like you can only add custom validation for single property values, not for complex logic based on multiple properties. 
Ill take another look and see how useful the failure is if it can at least be caught during `plan`

Reply:
you want the [new framework ](https://developer.hashicorp.com/terraform/plugin/framework/resources/validate-configuration)

Reply:
Yep once we are using that we will potentially have other options available.